src/features/search/pages/modals/VenueModal/useVenueModal.ts

```diff
@@ -0,0 +1,50 @@
+import { useState } from 'react'
+
+import { Venue } from 'features/venue/types'
+import { useDebounceValue } from 'ui/hooks/useDebounceValue'
+
+type VenueModalHook = {
+  doChangeVenue: (arg0: string) => void
+  doResetVenue: () => void
+  doSetSelectedVenue: (arg0: Venue) => void
+  isQueryProvided: boolean
+  shouldShowSuggestedVenues: boolean
+  isVenueNotSelected: boolean
+  venueQuery: string
+}
+
+const useVenueModal = (): VenueModalHook => {
+  const [venueQuery, setVenueQuery] = useState('')
+  const [selectedVenue, setSelectedVenue] = useState<Venue | null>(null)
+
+  const doChangeVenue = (text: string) => {
+    setVenueQuery(text)
+    setSelectedVenue(null)
+  }
+  const doResetVenue = () => {
+    setVenueQuery('')
+    setSelectedVenue(null)
+  }
+
+  const doSetSelectedVenue = (venue: Venue) => {
+    setSelectedVenue(venue)
+    setVenueQuery(venue.label)
+  }
+
+  const debouncedVenueQuery = useDebounceValue(venueQuery, 500)
+  const isQueryProvided = !!venueQuery && !!debouncedVenueQuery
+  const shouldShowSuggestedVenues = isQueryProvided && !selectedVenue
+  const isVenueNotSelected = !selectedVenue
```

Comment:
je trouve que les nom de boolean avec not dedans peuvent vite porter à confusion.
Je me retrouve souvent à faire confiance a cette article pour nommer mes variables https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig 
Et je trouve sont argument très bon si jamais quelqu'un veut réutiliser cette const il pourrait être amené à faire !isVenueNotSelected et la ça devient compliqué à comprendre.

Reply:
Done